### PR TITLE
feat: implements `usePollVote`

### PR DESCRIPTION
**Pull request by cesarenaldi**

### CODE REVIEW

File: packages/domain/src/entities/Signature.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import { Brand } from '@lens-protocol/shared-kernel';
+
+export type Signature = Brand<string, 'Signature'>;
```
Comment (reecejohnson) [inline packages/domain/src/entities/Signature.ts:3]: nice, love the explicit branding of the types here 👌

File: packages/react/src/polls/adapters/SnapshotVoteFactory.ts [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+}
+
+function isNumberArray(value: unknown): value is number[] {
+  return Array.isArray(value) && value.every(isNumber);
+}
+
+function validateMultipleChoice(
+  request: { choice: VoteChoice },
+  proposal: SnapshotProposal,
+): asserts request is { choice: number[] } {
```
Comment (reecejohnson) [inline packages/react/src/polls/adapters/SnapshotVoteFactory.ts:102]: clean 👍

File: packages/react/src/polls/adapters/types.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    { name: 'timestamp', type: 'uint64' },
+    { name: 'proposal', type: 'string' },
+    { name: 'choice', type: 'string' },
+    { name: 'reason', type: 'string' },
+    { name: 'app', type: 'string' },
+    { name: 'metadata', type: 'string' },
+  ],
+};
+
+export const vote2Types = {
```
[the remark this answers is not in the document]
Comment (cesarenaldi) [reply packages/react/src/polls/adapters/types.ts:40]: vote (no 2) is Snapshot v1 typed data... just their first iteration... v2 is the newer stuff. Anything you can create today on Snapshot uses the v2 versions.

File: packages/domain/src/use-cases/polls/__tests__/VotePoll.spec.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+      const presenter = mock<IVotePollPresenter>();
+
+      const { useCase } = setupTestScenario({ presenter, request });
+
+      await useCase.execute(request);
+
+      expect(presenter.present).toHaveBeenCalledWith(success());
+    });
+
+    xit('should present the error if the wallet fails to sign the vote', async () => {});
```
[the remark this answers is not in the document]
Comment (cesarenaldi) [reply packages/domain/src/use-cases/polls/__tests__/VotePoll.spec.ts:51]: mmmh, no good catch... I hope it's green

**Review by reecejohnson — APPROVED** [review]

**Pull request merged**